cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/namespace/NamespaceServiceTest.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    // "random" namespace should not exist
+    try {
+      namespaceService.getNamespace(Id.Namespace.from("random"));
+      Assert.fail("Namespace 'random' should not exist.");
+    } catch (NotFoundException e) {
+      Assert.assertEquals("Namespace", e.getElementType());
+      Assert.assertEquals("random", e.getElementId());
+    }
+
+    try {
+      try {
+        namespaceService.createNamespace(null);
+        Assert.fail("Namespace with null metadata should fail.");
+      } catch (IllegalArgumentException e) {
+        Assert.assertEquals("Namespace metadata cannot be null.", e.getMessage());
+      }
+
+      Assert.assertEquals(initialCount, namespaceService.listNamespaces().size());
+      verifyNotFound(namespaceId);
+
+      try {
+        namespaceService.createNamespace(builder.build());
+        Assert.fail("Namespace with no id should fail");
+      } catch (IllegalArgumentException e) {
+        Assert.assertEquals("Namespace id cannot be null.", e.getMessage());
+      }
+
+      Assert.assertEquals(initialCount, namespaceService.listNamespaces().size());
+      verifyNotFound(namespaceId);
+
+      try {
+        namespaceService.createNamespace(builder.setId(namespace).build());
+        Assert.fail("Namespace with no name should fail");
+      } catch (IllegalArgumentException e) {
+        Assert.assertEquals("Namespace name cannot be null.", e.getMessage());
+      }
+
+      Assert.assertEquals(initialCount, namespaceService.listNamespaces().size());
+      verifyNotFound(namespaceId);
+
+      try {
+        namespaceService.createNamespace(builder.setName(namespace).build());
+        Assert.fail("Namespace with no description should fail");
+      } catch (IllegalArgumentException e) {
+        Assert.assertEquals("Namespace description cannot be null.", e.getMessage());
+      }
+
+      try {
+        namespaceService.createNamespace(builder.setDescription("describes " + namespace).build());
+      } catch (IllegalArgumentException e) {
+        Assert.fail("Should successfully create a namespace.");
+      }
+    } catch (AlreadyExistsException e) {
+      Assert.fail(String.format("Namespace '%s' should not exist already.", namespace));
+    }
+
+    Assert.assertEquals(initialCount + 1, namespaceService.listNamespaces().size());
+
+    try {
+      NamespaceMeta namespaceMeta = namespaceService.getNamespace(namespaceId);
+      Assert.assertEquals(namespaceId.getId(), namespaceMeta.getId());
+      Assert.assertEquals(namespaceId.getId(), namespaceMeta.getName());
+      Assert.assertEquals("describes " + namespaceId.getId(), namespaceMeta.getDescription());
+
+      namespaceService.deleteNamespace(namespaceId);
+    } catch (NotFoundException e) {
+      Assert.fail(String.format("Namespace '%s' should be found since it was just created.", namespaceId.getId()));
+    }
+
+    verifyNotFound(namespaceId);
+  }
+
+  private static void verifyNotFound(Id.Namespace namespaceId) {
```

Comment:
The purpose of the method is to verify, no? It does so by using an ``assert``, but I feel ``assert`` is more of a language syntax thing. Prefer to call methods with their purposes in spoken (English) language (but don't feel to strongly, and can change).